src/server/game/Behavior/ActionScript/ActionThreadStep.cpp

```diff
@@ -29,15 +29,16 @@ class NullStep : public ActionThreadStep
         void Abort() override {}
 };
 
-/*static*/ std::unique_ptr<ActionThreadStep> ActionThreadStep::StepTo(ActionThread const& thread, ActionScriptStep const& stepTemplate)
+/*static*/ void ActionThreadStep::StepTo(unsigned char* ptr, ActionThread const& thread, ActionScriptStep const& stepTemplate)
```

Comment:
Is passing `ptr` really needed here, if you already have access to the thread? `&thread.GetCurrentStep()` and voila, no need for additional `reinterpret_cast`s either. Would require `thread` to not be `const`, of course.